### ISSUE TITLE
Support multiple return annoations

### ISSUE THREAD

**Issue by Crack** [EmmyLua/IntelliJ-EmmyLua#364]
Feature request: allow for multiple `@return` annotations. Supporting that would make some cases cleaner, eg. error handling, which is typically solved in Lua as returning "nil, error message".

Example
```lua
--- @param x number  Numeric parameter
--- @return number  Doubled value
--- @return nil, string  Error message
```

**Comment by LoganDark** [comment]
`--- @return (number) | (nil, string)`

**Comment by LordZZT** [comment]
I think what they are looking for is the ability to add the comments to the value specific annotation

**Comment by LoganDark** [comment]
@LordZZT You could include both cases in the one comment, which is what I do. Something like "Returns a number on success, or nil plus an error message on failure"

Otherwise, yes the feature doesn't currently exist in EmmyLua as far as I'm aware.

**Comment by Crack** [comment]
Yes, I would like to associate comments with return values. An example would be a method in simple HTTP API:

```lua
--- @param first_byte int  First byte of range (1-based)
--- @param byte_count int  Number of bytes to read
--- @return string|nil, nil|string, int|nil, int|nil
--- @return[1] string|nil  File bytes, nil if file doesn't exist
--- @return[1] nil
--- @return[1] int         File size
--- @return[1] int|nil     TTL, in seconds
--- @error Error message
function M:read_range(first_byte, byte_count)
```

Now, I know this can be put in description, but it's much easier to just skim over short descriptions of return values than parse entire method documentation looking for clues.

Your approach also looks unsupported:

```lua
--- @return (string, nil, int) | (nil, string)
```

is is highlighted this way:
![obraz](https://user-images.githubusercontent.com/1409807/93003521-b7959600-f53f-11ea-9dd8-fc9a259cd587.png)

So looks like the second result set is not expected here. And if I add a comment in this line it also doesn't show in parsed quick documentation window.
